<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2020-02-10" />

<title>Lassosum_analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">test_workflowr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/test_workflowr">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Lassosum_analysis</h1>
<h4 class="author"><em>Jing Gu</em></h4>
<h4 class="date"><em>2020-02-10</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PRS_project/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity itâ€™s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200205code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200205code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200205)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g.Â subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/projects/polygenic_risks/workbooks/PRS_project/output/%s_vs_wb_lm_summary.txt
</td>
<td style="text-align:left;">
output/%s_vs_wb_lm_summary.txt
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gutestworkflowrtree0f40724f26c86d3bd19393cdb5d69fcbcc1ee68ftargetblank0f40724a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/test_workflowr/tree/0f40724f26c86d3bd19393cdb5d69fcbcc1ee68f" target="_blank">0f40724</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gutestworkflowrtree0f40724f26c86d3bd19393cdb5d69fcbcc1ee68ftargetblank0f40724a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    output/new_results/

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/run_lassosum.R
    Untracked:  code/untitled.txt
    Untracked:  code/untitled1.txt
    Untracked:  data/gwas
    Untracked:  data/lassosum
    Untracked:  output/.ipynb_checkpoints/
    Untracked:  output/PRS_barplot.png
    Untracked:  output/PRSice_scatter_plot.png
    Untracked:  output/compare_tools.png
    Untracked:  output/lassosum_parameters.png

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If youâ€™ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/0f40724f26c86d3bd19393cdb5d69fcbcc1ee68f/analysis/Lassosum_analysis.Rmd" target="_blank">0f40724</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-24
</td>
<td>
add PRSice results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/test_workflowr/db9e0c202bee644246d795f9923fb08a2967beda/docs/Lassosum_analysis.html" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/3aa832ae9dd74b5b93c1c37b9ed799f5f67254a7/analysis/Lassosum_analysis.Rmd" target="_blank">3aa832a</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
<td>
run lassosum on blood-related traits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/d2b63a67acd41bb23de723d652ef0292379d9fe7/analysis/Lassosum_analysis.Rmd" target="_blank">d2b63a6</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-13
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/test_workflowr/d2b63a67acd41bb23de723d652ef0292379d9fe7/docs/Lassosum_analysis.html" target="_blank">d2b63a6</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-13
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/test_workflowr/8a24ca7ddf8df8dedceba9743609f55b4bdc4fa9/docs/Lassosum_analysis.html" target="_blank">8a24ca7</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-02-28
</td>
<td>
new
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/530f11ffe392f4852b40ea617f1e790686965d89/analysis/Lassosum_analysis.Rmd" target="_blank">530f11f</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-02-28
</td>
<td>
new
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/test_workflowr/373320f950baa65c3541a3171b11d2f3734d5f21/docs/Lassosum_analysis.html" target="_blank">373320f</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/5ab7e3b3143f2a70efe2c56cc9e7e1984849d9fd/analysis/Lassosum_analysis.Rmd" target="_blank">5ab7e3b</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-02-10
</td>
<td>
wflow_publish(â€œanalysis/*â€œ)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="construct-pgs-for-gtex-genotypes-using-lassosum" class="section level3">
<h3>Construct PGS for GTEx genotypes using Lassosum</h3>
<ul>
<li>Base dataset:
<ul>
<li>GIANT height 2014 (summary statistics)</li>
<li>CAD_2017: hg19</li>
</ul></li>
<li>Target datset:
<ul>
<li>GTEx whole blood</li>
</ul></li>
</ul>
</div>
<div id="overview-of-methods" class="section level3">
<h3>Overview of methods:</h3>
<ul>
<li>PGS constructed using summary statistics and a reference panel under a penalized regression framework (LASSO)</li>
<li>A general method for choosing tuning parameters without validation data</li>
<li>Comparable prediction accuracy to the dataset with phenotype available</li>
<li>Faster and more accurate than LDpred</li>
</ul>
</div>
<div id="qc-steps" class="section level3">
<h3>QC steps</h3>
<ul>
<li>base dataset
<ul>
<li>remove ambiguous SNPs (15%)</li>
<li>identify and remove duplicated SNPs (0%)</li>
</ul></li>
</ul>
</div>
<div id="target-dataset" class="section level3">
<h3>target dataset</h3>
<ol style="list-style-type: decimal">
<li>WTCCC
<ul>
<li>Human Genome Reference Assembly 17 (NCBI build 35)</li>
<li>Lift over to hg19</li>
</ul></li>
<li>GTEx Filtering Steps based on
<ul>
<li>filter snps by maf &gt;= 0.05, hwe &lt;1e-6, genotype missing &gt;= 0.01</li>
<li>filter inidividuals with &gt;= 0.01 genotype missingness (5 excluded)</li>
<li>remove highly correlated SNPs</li>
<li>remove individuals with high or low heterzygosity rates (16 excluded)</li>
<li>For the remaining samples, only <strong>654</strong> of them have genotype information for whole blood tissue.</li>
</ul></li>
</ol>
</div>
<div id="gene-expression-quantification" class="section level3">
<h3>Gene expression Quantification</h3>
<ul>
<li>Reads quantification read counts and TPM values were produced with RNA-SeQC v1.1.9*</li>
<li>Normalization (56200 genes -&gt; 32387 genes)
<ol style="list-style-type: decimal">
<li>Trimmed mean of M-values (TMM) between each pair of samples {<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25</a>}
<ul>
<li>address the problem of a few genes expressed way more higher in one sample than the rest. Other genes may seem to be downregulated compared with spurious expression of a small number of genes</li>
<li>compute a set of scaling factors for the library sizes that minimize the log-fold changes between the pair of samples for most genes.</li>
</ul></li>
<li>Filtered genes by gene expression &gt;= 0.1 in at least 20% of samples</li>
<li>Expression values of each gene inversely normal transformed across samples</li>
</ol>
<ul>
<li>rank-based INT</li>
<li>convert to ranks, then transform to quantiles</li>
<li>sample from inverse normal distributions</li>
</ul></li>
<li>Plots for Normalization</li>
</ul>
<pre><code>No id variables; using all as measure variables</code></pre>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-1-1.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/1b09763dae4a1e5dcf975c722e2913b747071f59/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-1-1.png" target="_blank">1b09763</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Check distributinos for specific genesâ€™s transcripts levels <img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-2-2.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Try log2 transformation <img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-3-2.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Covariates:
<ol style="list-style-type: decimal">
<li>first five genotype PCs as covariates, which capture the major population structure among GTEx donors.</li>
<li>WGS sequencing platform and library protocol</li>
<li>donor sex, age, ethnicity</li>
</ol></li>
</ul>
<p><img src="../output/GTEx_PC_covariates.png" alt="PCA results for GTEx" /> ### Results</p>
<ul>
<li>Evaluate the performance of lassosum and PRSice-2 on WTCCC dataset <img src="../output/compare_tools.png" alt="Scatter plots for true phenotypic values against PRS" /> <img src="../output/PRS_barplot.png" alt="Performance of PRSice" /> <img src="../output/PRSice_scatter_plot.png" alt="Performance of PRSice" /></li>
<li>Results from GTEx <img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/test_workflowr/blob/db9e0c202bee644246d795f9923fb08a2967beda/docs/figure/Lassosum_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">db9e0c2</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-03-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;The number of significant associations for the trait after FDR correction is 0&quot;</code></pre>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;The number of significant associations for the trait after FDR correction is 0&quot;</code></pre>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;The number of significant associations for the trait after FDR correction is 0&quot;</code></pre>
</div>
<div id="gene-ac080125.1-has-the-strongest-association-with-cd-trait" class="section level3">
<h3>Gene AC080125.1 has the strongest association with CD trait</h3>
<p><img src="figure/Lassosum_analysis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reshape2_1.4.3 ggplot2_3.1.1 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2       compiler_3.5.1   pillar_1.3.1     later_0.7.5     
 [5] git2r_0.26.1     highr_0.7        plyr_1.8.4       workflowr_1.6.0 
 [9] tools_3.5.1      digest_0.6.18    evaluate_0.12    tibble_2.1.1    
[13] gtable_0.2.0     pkgconfig_2.0.2  rlang_0.4.0      yaml_2.2.0      
[17] withr_2.1.2      stringr_1.3.1    dplyr_0.8.0.1    knitr_1.20      
[21] fs_1.3.1         rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[25] glue_1.3.0       R6_2.3.0         rmarkdown_1.10   purrr_0.3.2     
[29] magrittr_1.5     whisker_0.3-2    backports_1.1.2  scales_1.0.0    
[33] promises_1.0.1   htmltools_0.3.6  assertthat_0.2.0 colorspace_1.3-2
[37] httpuv_1.4.5     labeling_0.3     stringi_1.2.4    lazyeval_0.2.1  
[41] munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
